arches/app/models/models.py

```diff
@@ -1152,6 +1197,10 @@ def __init__(self, *args, **kwargs):
         if not self.tileid:
             self.tileid = uuid.uuid4()
 
+    @property
+    def nodegroup(self):
```

Comment:
Wondering if you considered making this a cached property (given how unlikely noedgroup-deletions-in-production are)?

> Be careful with your own custom properties - it is up to you to implement caching when required, for example using the [cached_property](https://docs.djangoproject.com/en/4.2/ref/utils/#django.utils.functional.cached_property) decorator.

https://docs.djangoproject.com/en/4.2/topics/db/optimization/#understand-cached-attributes

Reply:
My gut says to shy away from caching in this instance because of the potential for unexpected behavior during graph-switching or updating

Reply:
sounds wise 👍